src/gui/widgets/TabWidget.cpp

```diff
@@ -192,24 +253,66 @@ void TabWidget::paintEvent( QPaintEvent * _pe )
 	}
 	p.setPen( cap_col );
 
-
-	for( widgetStack::iterator it = m_widgets.begin();
-						it != m_widgets.end(); ++it )
+	// Compute tabs' width depending on the number of tabs (only applicable for artwork tabs)
+  	widgetStack::iterator first = m_widgets.begin();
+  	widgetStack::iterator last = m_widgets.end();
+	int tab_width = width();
+	if ( first != last )
 	{
-		if( it.key() == m_activeTab )
+		tab_width = ( width() - tab_x_offset ) / std::distance( first, last );
+	}
+
+	// Draw all tabs
+        for( widgetStack::iterator it = first ; it != last ; ++it )
+        {
+
+		// Draw a text tab or a artwork tab.
+		if ( m_usePixmap )
+		{	
+
+			// Fixes tab's width, because original size is only correct for text tabs
+			( *it ).nwidth = tab_width;
+
+			// Get active or inactive artwork
+                	QPixmap *artwork;
+			if( it.key() == m_activeTab )
+                        {
+	                	artwork = new QPixmap( embed::getIconPixmap( ( *it ).activePixmap ) );
+				p.fillRect( tab_x_offset, 2, ( *it ).nwidth, m_tabbarHeight - 3, cap_col );
+			} else 
+			{
+	                	artwork = new QPixmap( embed::getIconPixmap( ( *it ).inactivePixmap ) );
+			}
+
+			// Draw artwork
+	                p.drawPixmap(tab_x_offset + ( ( *it ).nwidth - ( *artwork ).width() ) / 2, 2, *artwork );
+
+		} else
 		{
-			p.setPen( QColor( 32, 48, 64 ) );
-			p.fillRect( tab_x_offset, 2, ( *it ).nwidth - 6, 10, cap_col );
+
+			// Highlight tab when active
+                	if( it.key() == m_activeTab )
+	                {
+				p.setPen( QColor( 32, 48, 64 ) );
```

Comment:
This color should not be hardcoded, and you'd want to make this themeable via CSS.